content/docs/functions/json_build_object.md

```diff
@@ -0,0 +1,265 @@
+       )
+   ) AS details
+FROM products;
+```
+
+
+Returns:
+
+```text
+| id |    name     | price |                                                               details                                                              
+|----|-------------|-------|-------------------------------------------------------------------------------------------------------------------------------------
+| 1  | T-Shirt     | 25.99 | {"category" : "Clothing", "description" : "A comfortable cotton T-Shirt", "attributes" : {"color" : "Blue", "size" : "Medium"}}
+| 2  | Coffee Mug  | 12.99 | {"category" : "Kitchen", "description" : "A ceramic mug with a funny design", "attributes" : {"color" : "White", "size" : "Large"}}
+```
+
+### Use with `ORDER BY`
+
+
+Combine `json_build_object` with `ORDER BY` to sort the results based on a specific attribute within the JSON structure.
+
+
+Let's consider an example with a modified version of the `products` table where each product has a `rating` attribute:
+
+
+**products**
+
+```text
+| id |    name    | price |            description            | category |                     attributes                    
+|----|------------|-------|-----------------------------------|----------|----------------------------------------------------
+| 1  | T-Shirt    | 25.99 | A comfortable cotton T-Shirt      | Clothing | {"size": "Medium", "color": "Blue", "rating": 4.5}
+| 2  | Coffee Mug | 12.99 | A ceramic mug with a funny design | Kitchen  | {"size": "Small", "color": "White", "rating": 3.8}
+| 3  | Sneakers   | 49.99 | Sporty sneakers for everyday use  | Footwear | {"size": "10", "color": "Black", "rating": 4.2}
+```
+
+{/*
+```sql
+CREATE TABLE products_with_rating (
+   id SERIAL PRIMARY KEY,
+   name TEXT NOT NULL,
+   price DECIMAL(5, 2) NOT NULL,
+   description TEXT,
+   category TEXT,
+   attributes JSONB
+);
+
+
+INSERT INTO products_with_rating (name, price, description, category, attributes)
+VALUES
+   ('T-Shirt', 25.99, 'A comfortable cotton T-Shirt', 'Clothing', json_build_object(
+       'color', 'Blue',
+       'size', 'Medium',
+       'rating', 4.5
+   )),
+   ('Coffee Mug', 12.99, 'A ceramic mug with a funny design', 'Kitchen', json_build_object(
+       'color', 'White',
+       'size', 'Small',
+       'rating', 3.8
+   )),
+   ('Sneakers', 49.99, 'Sporty sneakers for everyday use', 'Footwear', json_build_object(
+       'color', 'Black',
+       'size', '10',
+       'rating', 4.2
+   ));
+```
+*/}
+
+
+Retrieve the products, building a JSON structure with `json_build_object` and ordering the results based on the `rating`:
+
+
+```sql
+SELECT
+   id,
+   name,
+   price,
+   json_build_object(
+       'category', category,
+       'description', description,
+       'attributes', json_build_object(
+           'color', attributes->>'color',
+           'size', attributes->>'size',
+           'rating', attributes->>'rating'
+       )
+   ) AS details
+FROM products_with_rating
```

Comment:
Small thing, but I did find this confusing at first to see this 2nd 'products_with_rating' table. See comment above.